### PR TITLE
Temp heap usage reduction

### PR DESCRIPTION
**Pull request by TeroJaasko**

### CODE REVIEW

**Comment by TeroJaasko** [comment]
@yogpan01 @kuggenhoffen @simosillankorva @anttiylitokola

**Comment by yogpan01** [comment]
+1, lets merge this now and run some regression to check if there are some issues

File: source/m2mstrinbufferbase.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "m2mstringbufferbase.h"
+
+#include "m2mstring.h"
```
Comment (teetak01) [inline source/m2mstrinbufferbase.cpp]: Should these be "mbed-client/xxx.h" for yotta-builds?

File: mbed-client/m2mstringbuffer.h [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __STRING_BUFFER_H__
+#define __STRING_BUFFER_H__
+
+#include "m2mstringbufferbase.h"
```
Comment (teetak01) [inline mbed-client/m2mstringbuffer.h]: probably also here "mbed-client/m2mstringbufferbase.h"

**Comment by TeroJaasko** [comment]
@teetak01 good catch. I just rebased with current base branch did the proposed changes and squashed them in.

**Pull request merged**